weather_dl/download_pipeline/partition.py

```diff
@@ -137,8 +153,14 @@ def prepare_partitions(config: Config) -> t.Iterator[Config]:
     Returns:
         An iterator of `Config`s.
     """
-    for option in itertools.product(*[config.selection[key] for key in config.partition_keys]):
-        yield _create_partition_config(option, config)
+    if config.client == 'eumetsat':
+        client = CLIENTS[config.client](config)
+        slices = time_slices(config.selection['start_date'], config.selection['end_date'])
+        for option in client.products(slices, config.dataset):
+            yield _create_partition_config((option, ), config)
```

Comment:
Does this run in parallel for each slice? Querying the products can take some time, so this should run in parallel on several workers as part of the pipeline.

Reply:
Necessary code changes done. Thanks.